src/main/java/org/jalasoft/BankAccount.java

```diff
@@ -0,0 +1,122 @@
+package org.jalasoft;
+
+/**
+ * 
+ */
+public class BankAccount {
+
+    private int accountNumber;
+    private AccountOrigin accountOrigin;
+    private int balance;
+
+    /**
+     * Create a new account with 0 balance
+     * 
+     * @param accountNumber the unique identifier for a bank account
+     * @param accountOrigin represent where the account was created
+     */
+    public BankAccount(int accountNumber, AccountOrigin accountOrigin) {
+        this.accountNumber = accountNumber;
+        this.accountOrigin = accountOrigin;
+        this.balance = 0;
+    }
+
+    /**
+     * @return the unique identifier of the account
+     */
+    public int getAccountNumber() {
+        return accountNumber;
+    }
+
+    /**
+     * Return the current balance of a given account
+     * 
+     * @return the balance of the given account
+     */
+    public int getBalance() {
+        return balance;
+    }
+
+    /**
+     * @return the origin where the account was created
+     */
+    public AccountOrigin getAccountOrigin() {
+        return accountOrigin;
+    }
+
+    /**
+     * This increase the amount of the balance applying the following restrictions:
+     *  The amount must be greater than 0
+     * 
+     * @param amount the amount that will be increased
+     * @throws IllegalArgumentException will be throw if the amount is less or equals than 0
+     */
+    public void deposit(int amount) throws IllegalArgumentException {
+        if (amount <= 0) {
+            throw new IllegalArgumentException("The deposit amount must be greater than 0");
+        }
+
+        balance += amount;
+    }
+
+    /**
+     * This decrease the amount of the balance applying the following restrictions:
+     * - The amount cannot be greater than the current balance
+     * 
+     * @param amount the amount that will be decrease
+     */
+    public void withdraw(int amount) throws IllegalArgumentException {
+      /*   if (amount <= 0) {
+            throw new IllegalArgumentException("The amount to withdraw must be greater than 0.");
+        }
+
+       if (amount > balance) {
+            throw new ArithmeticException("Cannot withdraw amounts greater than the actual balance: " + getBalance());
+        }*/
+
+        
+      
+        try {
+            if (amount <= 0) {
+                throw new IllegalArgumentException("The amount to withdraw must be greater than 0.");
+            }
+            if (amount > balance) {
+                throw new ArithmeticException("Cannot withdraw amounts greater than the actual balance: " + getBalance());}
+            
+                balance -= amount;
+           
+        } catch (ArithmeticException exception) {
+            System.out.println(exception.getMessage());
```

Comment:
Nice try!!, but remember that bank services is part of the business logic which is not responsible for user interaction so it means you cannot use `System.out.println`.

So here you should let the exceptions throw and then catch them in the BankClient